rest/tests/test_repository.py

```diff
@@ -0,0 +1,322 @@
+
+        # put with missing slug
+        self.put_repository("missing slug", input_dict,
+                            expected_status=HTTP_404_NOT_FOUND)
+
+        self.put_repository(
+            repo_slug, input_dict,
+            expected_status=HTTP_405_METHOD_NOT_ALLOWED)
+
+        # verify change of data
+        self.put_repository(
+            repo_slug, input_dict_changed,
+            expected_status=HTTP_405_METHOD_NOT_ALLOWED)
+
+        # never created a duplicate
+        self.assertEqual(2, Repository.objects.count())
+
+        # call to verify HTTP_OK
+        self.get_repository(repo_slug)
+
+        # delete a repository
+        self.delete_repository(repo_slug,
+                               expected_status=HTTP_405_METHOD_NOT_ALLOWED)
+
+        repositories = self.get_repositories()
+        self.assertEqual(2, repositories['count'])
+
+        vocab_slug = self.create_vocabulary(self.repo.slug)['slug']
+        vocabularies = self.get_vocabularies(self.repo.slug)
+        self.assertEqual(1, vocabularies['count'])
+
+        self.delete_repository(self.repo.slug,
+                               expected_status=HTTP_405_METHOD_NOT_ALLOWED)
+
+        self.delete_vocabulary(self.repo.slug, vocab_slug)
+
+        vocabularies = self.get_vocabularies(self.repo.slug)
+        self.assertEqual(0, vocabularies['count'])
+
+        self.delete_repository(self.repo.slug,
+                               expected_status=HTTP_405_METHOD_NOT_ALLOWED)
+
+        self.get_repositories()
+        self.get_repository("missing", expected_status=HTTP_404_NOT_FOUND)
+
+        # as anonymous
+        self.logout()
+        self.get_repositories(expected_status=HTTP_403_FORBIDDEN)
+        self.get_repository("missing", expected_status=HTTP_403_FORBIDDEN)
+
+    def test_repository_pagination(self):
+        """Test pagination for collections"""
+
+        expected = [
+            self.create_repository(
+                {
+                    'name': "name{i}".format(i=i),
+                    "description": "description"
+                }
+            ) for i in range(40)]
+
+        repositories = self.get_repositories()
+
+        # 40 we created + self.repo
+        self.assertEqual(41, repositories['count'])
+        self.assertEqual(
+            [RepositorySerializer(
+                Repository.objects.get(id=x['id'])).data
+             for x in expected[:19]],
+            repositories['results'][1:20])
+
+        resp = self.client.get(
+            '{repo_base}?page=2'.format(
+                repo_base=REPO_BASE,
+            ))
+        self.assertEqual(HTTP_200_OK, resp.status_code)
+        repositories = as_json(resp)
+        self.assertEqual(41, repositories['count'])
+        self.assertEqual(
+            [RepositorySerializer(Repository.objects.get(id=x['id'])).data
+             for x in expected[19:39]], repositories['results'])
+
+    # for some reason Pylint has issues with this name
```

Comment:
the issue is the length of the name (I guess), but in the tests we can even disable this globally